vxquery-core/src/main/java/org/apache/vxquery/xmlparser/IParser.java

```diff
@@ -25,5 +25,7 @@
 public interface IParser {
     public void parseDocument(File file, ArrayBackedValueStorage abvs) throws HyracksDataException;
 
+    public void parseString(String input, ArrayBackedValueStorage abvs) throws HyracksDataException;
+
```

Comment:
Since we have 3 methods on IParser with 2 methods not being implemented by at least one parser, I think that we should just have one method on the interface that takes an java.io.Reader. We then could implement the other signatures by delegating to the one method on IParser.